packages/system/src/Stream/Stream/filter.ts

```diff
@@ -0,0 +1,17 @@
+import * as Array_ from "../../Array"
+import type { Predicate } from "../../Function"
+import { pipe } from "../../Function"
+import type { Stream } from "./definitions"
+import { mapChunks } from "./mapChunks"
+
+/**
+ * Applies the predicate to each element and allows passing elements 
+ * to reach the output of this stream.
+ */
+export const filter = <O>(f: Predicate<O>) => <R, E>(
+  self: Stream<R, E, O>
+): Stream<R, E, O> =>
+  pipe(
```

Comment:
Looks like this pipe is not necessary.

Reply:
happy to make this change - original thought was to model it after mapConcat which uses pipe and Array_.chain_